cou/utils/openstack.py

```diff
@@ -147,3 +209,117 @@ def lookup(cls, component: str, version: str) -> List[str]:
             if version in version_range:
                 compatible_os_releases.append(openstack_release)
         return compatible_os_releases
+
+
+class OpenStackRelease:
+    """Provides a class that will compare strings from an iterator type object.
+
+    Used to provide > and < comparisons on strings that may not necessarily be
+    alphanumerically ordered.  e.g. OpenStack or Ubuntu releases AFTER the
+    z-wrap.
+    """
+
+    openstack_codenames = list(OPENSTACK_CODENAMES.keys())
+    openstack_release_year = list(OPENSTACK_CODENAMES.values())
+
+    def __init__(self, os_release: str):
+        """Initialize the OpenStackRelease object.
+
+        :param os_release: OpenStack release codename.
+        :type os_release: str
+        :raises ValueError: Raises ValueError if OpenStack codename is unknown.
+        """
+        self.os_release = os_release
+
+    def __eq__(self, other):
+        """Do equals."""
+        if not isinstance(other, (str, OpenStackRelease)):
+            return NotImplemented
+        return self.index == self.openstack_codenames.index(other)
+
+    def __ne__(self, other):
+        """Do not equals."""
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        """Do less than."""
+        if not isinstance(other, (str, OpenStackRelease)):
+            return NotImplemented
+        return self.index < self.openstack_codenames.index(other)
+
+    def __ge__(self, other):
+        """Do greater than or equal."""
+        return not self.__lt__(other)
+
+    def __gt__(self, other):
+        """Do greater than."""
+        if not isinstance(other, (str, OpenStackRelease)):
+            return NotImplemented
+        return self.index > self.openstack_codenames.index(other)
+
+    def __le__(self, other):
+        """Do less than or equals."""
+        return not self.__gt__(other)
+
+    def __repr__(self):
+        """Return the representation of CompareOpenStack."""
+        return f"{self.__class__.__name__}<{self.os_release}>"
+
+    @property
+    def os_release(self) -> str:
```

Comment:
`OpenStackRelease.os_release` seems quite redundant. How about simply `codename`?
```suggestion
    def codename(self) -> str:
```
You'd then have a more natural UX
```
my_openstack = OpenStackRelease('ussuri')
print(my_openstack.codename) # prints 'ussuri'
```